pkg/addonmanager/manager.go

```diff
@@ -183,34 +271,26 @@ func (a *addonManager) Start(ctx context.Context) error {
 		addonConfigController = addonconfig.NewAddonConfigController(
 			addonClient,
 			addonInformers.Addon().V1alpha1().ManagedClusterAddOns(),
+			addonInformers.Addon().V1alpha1().ClusterManagementAddOns(),
 			dynamicInformers,
-			a.addonConfigs,
+			utils.FilterOutTheBuiltInAddOnConfigGVRs(a.addonConfigs),
```

Comment:
May need to consider the case when the addon upgrades the framework but still wants to be managed by self, which component is responsible to update the spec hash and lastobservedgeneration for addondeploymentconfig?

Reply:
deleted the addonconfig controller in the addon manager, will use the controller in addon-framework to update the spec hash for both addon templates and addon deployment config.